### PR TITLE
Restrict access in home content block

### PR DESCRIPTION
**Pull request by LauraJaimeC**
#### :tophat: What? Why?

- [x] Assemblies
- [x] Consultations
- [x] Participatory processes

#### :clipboard: Subtasks
- [ ] Add tests
- [ ] Add documentation regarding the feature 
- [ ] Add/modify seeds
- [ ] Another subtask

### :camera: Screenshots (optional)
Description

### CODE REVIEW

File: app/decorators/cells/decidim/assemblies/content_blocks/highlighted_assemblies_cell_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# This decorator filter assemblies by permissions in home content block.
+Decidim::Assemblies::ContentBlocks::HighlightedAssembliesCell.class_eval do
+  include FilterParticipatorySpacesHelper
+
+  def highlighted_assemblies
+    @highlighted_assemblies ||= if current_user.admin?
+                                  Decidim::Assemblies::OrganizationPrioritizedAssemblies
```
Comment (tramuntanal) [inline app/decorators/cells/decidim/assemblies/content_blocks/highlighted_assemblies_cell_decorator.rb:9]: Usually if we want the standard behavior we delegate to the decorated method. Check this example: https://github.com/gencat/participa-catalunya/blob/master/app/decorators/decidim/courses/controllers/admin/courses_controller_decorator.rb

File: spec/cells/decidim/assemblies/content_blocks/highlighted_assemblies_cell_spec.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        let(:content_block) { create :content_block, organization: organization, manifest_name: :highlighted_assemblies, scope_name: :homepage, settings: settings }
+        let(:current_user) { create(:user, :confirmed, organization: organization) }
+        let(:metadata) { { ceco: "ceco", ceco_txt: "ceco_txt" } }
+        let!(:authorization) { create(:authorization, user: current_user, name: "dummy_authorization_handler", metadata: metadata) }
+
+        let!(:assembly_with_filter) { create(:assembly, :published, :promoted, organization: organization, suara_permissions: { ceco: "ceco", ceco_txt: "ceco_txt" }) }
+        let!(:assembly_with_other_filter) { create(:assembly, :published, :promoted, organization: organization, suara_permissions: { ceco: "a", ceco_txt: "b" }) }
+        let!(:assembly_without_filter) { create(:assembly, :published, :promoted, organization: organization, suara_permissions: { ceco: "", ceco_txt: "" }) }
+
+        controller Decidim::PagesController
```
Comment (tramuntanal) [inline spec/cells/decidim/assemblies/content_blocks/highlighted_assemblies_cell_spec.rb:21]: Shouldn't it be `homepage_controller`?

File: spec/cells/decidim/consultations/content_blocks/highlighted_consultations_cell_spec.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        let(:content_block) { create :content_block, organization: organization, manifest_name: :highlighted_consultations, scope_name: :homepage, settings: settings }
+        let(:current_user) { create(:user, :confirmed, organization: organization) }
+        let(:metadata) { { ceco: "ceco", ceco_txt: "ceco_txt" } }
+        let!(:authorization) { create(:authorization, user: current_user, name: "dummy_authorization_handler", metadata: metadata) }
+
+        let!(:consultation_with_filter) { create(:consultation, :active, organization: organization, suara_permissions: { ceco: "ceco", ceco_txt: "ceco_txt" }) }
+        let!(:consultation_with_other_filter) { create(:consultation, :active, organization: organization, suara_permissions: { ceco: "a", ceco_txt: "b" }) }
+        let!(:consultation_without_filter) { create(:consultation, :active, organization: organization, suara_permissions: { ceco: "", ceco_txt: "" }) }
+
+        controller Decidim::PagesController
```
Comment (tramuntanal) [inline spec/cells/decidim/consultations/content_blocks/highlighted_consultations_cell_spec.rb:21]: Shouldn't it be `homepage_controller`?

File: spec/cells/decidim/participatory_processes/content_blocks/highlighted_processes_cell_spec.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        let(:content_block) { create :content_block, organization: organization, manifest_name: :highlighted_processes, scope_name: :homepage, settings: settings }
+        let(:current_user) { create(:user, :confirmed, organization: organization) }
+        let(:metadata) { { ceco: "ceco", ceco_txt: "ceco_txt" } }
+        let!(:authorization) { create(:authorization, user: current_user, name: "dummy_authorization_handler", metadata: metadata) }
+
+        let!(:participatory_process_with_filter) { create(:participatory_process, :published, :promoted, organization: organization, suara_permissions: { ceco: "ceco", ceco_txt: "ceco_txt" }) }
+        let!(:participatory_process_with_other_filter) { create(:participatory_process, :published, :promoted, organization: organization, suara_permissions: { ceco: "a", ceco_txt: "b" }) }
+        let!(:participatory_process_without_filter) { create(:participatory_process, :published, :promoted, organization: organization, suara_permissions: { ceco: "", ceco_txt: "" }) }
+
+        controller Decidim::PagesController
```
Comment (tramuntanal) [inline spec/cells/decidim/participatory_processes/content_blocks/highlighted_processes_cell_spec.rb:21]: Shouldn't it be `homepage_controller`?

**Review by tramuntanal — CHANGES_REQUESTED** [review]
Good job @LauraJaimeC !

I've left you a recommendation to be applied to all decorators so that if the decorated command changes, the decorators are not affected.

**Comment by LauraJaimeC** [comment]
Changes applied!

**Review by tramuntanal — DISMISSED** [review]
LGTM!

**Pull request merged**